### ISSUE TITLE
[Feature request] recipe only based on reached milestone

### ISSUE THREAD

**Issue by LoneWolfProductions** [ShadowTheAge/yafc#120]
Option to only show current recipes based on milestones selected/reached
(if a milestone/additional research is not reached/needed leave it out to avoid clutter)

this would be nice if possible because some modded games have alot of possibilities in the future that you can't do yet.

Highlight the best option (yafc costs still confusses me till this day :p)

**Comment by ShadowTheAge** [comment]
Where exactly? In the NEIE they are already under a button, and in productiontable dropdowns they are sorted down

Also milestones are more precise than "additional research"
If a recipe has an ingredient locked behind a milestone, the whole recipe will be shown as requiring a milestone. While it may be researchable and selectable in game. hiding those is really bad idea.

What clutter are you trying to avoid? Recipes requiring an unreached milestone are clearly marked as so.

**Comment by LoneWolfProductions** [comment]
Ok i found the option in NEIE, and if i understand correctly then 

In the "add production recipe" tab  they are sorted from best to worst (at the current state of the milestones)?
Followed by locked behind milestones?

if so then i rest my case, i know what is the best option at this time without hovering over all possibilties trying to figure out what to choose as best option (unless desired)

the clutter happens sometimes when clicking "see full list" (see image) i just want to know the best option :p 

![image](https://user-images.githubusercontent.com/30441722/119853944-ee1a1200-bf10-11eb-8c0c-d343b6d6c1f6.png)

**Comment by ShadowTheAge** [comment]
Yes, the first sorting priority is global availibility, the second is milestone availibility, and the third is what yafc thinks best.
